charts/turbinia/README.md

```diff
@@ -41,44 +40,197 @@ helm repo add osdfir-charts https://google.github.io/osdfir-infrastructure/
+of the image using the `image.tag` parameter and/or a different repository using
+the `image.repository` parameter.
 
-Pull the chart locally then cd into `/turbinia` and review the `values-production.yaml` file for a list of values that will be used for production.
+For example, to use the most recent development
+version instead, set the following variables:
 
 ```console
-helm pull osdfir-charts/turbinia --untar
+turbinia.server.image.repository="us-docker.pkg.dev/osdfir-registry/turbinia/release/turbinia-server-dev"
+turbinia.server.image.tag="latest"
+turbinia.api.image.repository="us-docker.pkg.dev/osdfir-registry/turbinia/release/turbinia-api-server-dev"
+turbinia.api.image.tag="latest"
+turbinia.worker.image.repository="us-docker.pkg.dev/osdfir-registry/turbinia/release/turbinia-worker-dev"
+turbinia.worker.image.tag="latest"
 ```
 
-### GKE Installations
+### Upgrading the Helm chart
+
+Helm chart updates can be retrieved by running `helm repo update`.
+
+To explore available charts and versions, use `helm search repo osdfir-charts/`.
+Install a specific chart version with `helm install my-release osdfir-charts/turbinia --version <version>`.
+
+A major Helm chart version change (like v1.0.0 -> v2.0.0) indicates that there
+is an incompatible breaking change needing manual actions.
+
+### Managing and updating the Turbinia config
+
+This section outlines how to deploy and manage the Turbinia configuration file
+within OSDFIR infrastructure.
+
+There are two primary methods:
+
+#### Using Default Configurations
 
-Create a Turbinia GCP account using the helper script in `tools/create-gcp-sa.sh` prior to installing the chart.
+If you don't provide your own Turbinia config file during deployment,
+the Turbinia deployment will automatically retrieve the latest default configs
+from the Turbinia Github repository. This method requires no further action from you.
+
+> **NOTE:**  When using the default method, you cannot update the Turbinia config file directly.
+
+#### Managing Turbinia configs externally
+
+For more advanced configuration management, you can manage the Turbinia config
+file independently of the Helm chart:
+
+1. Prepare your Config File:
+
+    Organize the Turbinia config file with your desired customizations.
+
+2. Create a ConfigMap:
+
+    ```console
+    kubectl create configmap turbinia-configs --from-file=turbinia.conf
+    ```
+
+    Replace `turbinia.conf` with the actual name of your config file.
+
+3. Install or Upgrade the Helm Chart:
+
+    ```console
+    helm install my-release osdfir-charts/turbinia --set config.existingConfigMap="turbinia-configs"
+    ```
+
+    This command instructs the Helm chart to use the `turbinia-configs` ConfigMap for
+    Turbinia's config file.
+
+To update the config changes using this method:
+
+1. Update the ConfigMap:
+
+    ```console
+    kubectl create configmap turbinia-configs --from-file=turbinia.conf --dry-run -o yaml | kubectl replace -f -
```

Comment:
Noo it's because without `--dry-run` it will complain that the configmap exists and there isn't any direct `update` you can use so you re-create the yaml file with the --dry-run and replace the ConfigMap file containing the turbina config. A little convoluted.. yes